com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -46,34 +59,36 @@ internal NetworkSerializer __beginSendServerRpc(uint rpcMethodId, ServerRpcParam
         {
             PooledNetworkWriter writer;
 
-            var rpcQueueContainer = NetworkManager.RpcQueueContainer;
-            var isUsingBatching = rpcQueueContainer.IsUsingBatching();
+            SetUpdateStage(ref serverRpcParams.Send);
+
+            if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
+            {
+                throw new NotSupportedException(
+                    $"{nameof(NetworkUpdateStage.Initialization)} cannot be used as a target for processing RPCs.");
```

Comment:
```suggestion
                    $"{SendServerRPC failed: update stage cannot be in nameof(NetworkUpdateStage.Initialization)} state.");
```

Reply:
As I was about to apply this change, it occurred to me that the new message is actually less accurate. If UpdateStage is set to Initialization after the call to `SetUpdateStage()`, it means the user passed in an invalid value - they explicitly stated as a parameter that they wanted it to be Initialization. So that doesn't mean the update stage is IN initialization, it means the user REQUESTED initialization.

Reply:
👍🏽